packages/web/src/components/track/DownloadRow.tsx

```diff
@@ -16,35 +27,43 @@ const messages = {
 }
 
 type DownloadRowProps = {
-  trackId: ID
   onDownload: (args: { trackIds: ID[]; parentTrackId?: ID }) => void
+  isOriginal: boolean
+  trackId?: ID
   hideDownload?: boolean
   index: number
   size?: number
-  isOriginal: boolean
+  category?: StemCategory
+  filename?: string
+  isLoading?: boolean
 }
 
 export const DownloadRow = ({
-  trackId,
   onDownload,
+  isOriginal,
+  trackId,
   hideDownload,
   index,
   size,
-  isOriginal
+  category,
+  filename,
+  isLoading
 }: DownloadRowProps) => {
   const track = useSelector(
     (state: CommonState) => getTrack(state, { id: trackId }),
     shallowEqual
   )
-  const { shouldDisplayDownloadFollowGated } = useDownloadableContentAccess({
-    trackId
-  })
+  const { shouldDisplayDownloadFollowGated } = trackId
+    ? useDownloadableContentAccess({
+        trackId
+      })
+    : { shouldDisplayDownloadFollowGated: false }
 
   const downloadButton = () => (
     <PlainButton
       onClick={() =>
         onDownload({
-          trackIds: [trackId]
+          trackIds: trackId ? [trackId] : []
```

Comment:
hm why did `trackId` become optional? these ternaries are a bit hard to read.

Reply:
ohh it's to support stems that are currently uploading that aren't persisted yet. gotcha.

Reply:
because in the edit flow, uploading/pending stems dont have track ids